x-pack/plugins/lens/public/embeddable/embeddable_factory.ts

```diff
@@ -91,57 +95,61 @@ export class EmbeddableFactory implements EmbeddableFactoryDefinition {
   };
 
   async create(input: LensEmbeddableInput, parent?: IContainer) {
-    const {
-      data,
-      timefilter,
-      expressionRenderer,
-      documentToExpression,
-      injectFilterReferences,
-      visualizationMap,
-      datasourceMap,
-      uiActions,
-      coreHttp,
-      attributeService,
-      indexPatternService,
-      capabilities,
-      usageCollection,
-      theme,
-      inspector,
-      spaces,
-      uiSettings,
-    } = await this.getStartServices();
-
-    const { Embeddable } = await import('../async_services');
-
-    return new Embeddable(
-      {
-        attributeService,
+    try {
+      const {
         data,
-        indexPatternService,
         timefilter,
-        inspector,
         expressionRenderer,
-        basePath: coreHttp.basePath,
-        getTrigger: uiActions?.getTrigger,
-        getTriggerCompatibleActions: uiActions?.getTriggerCompatibleActions,
         documentToExpression,
         injectFilterReferences,
         visualizationMap,
         datasourceMap,
-        capabilities: {
-          canSaveDashboards: Boolean(capabilities.dashboard?.showWriteControls),
-          canSaveVisualizations: Boolean(capabilities.visualize.save),
-          navLinks: capabilities.navLinks,
-          discover: capabilities.discover,
-        },
+        uiActions,
+        coreHttp,
+        attributeService,
+        indexPatternService,
+        capabilities,
         usageCollection,
         theme,
+        inspector,
         spaces,
         uiSettings,
-      },
-      input,
-      parent
-    );
+      } = await this.getStartServices();
+
+      const { Embeddable } = await import('../async_services');
+
+      return new Embeddable(
+        {
+          attributeService,
+          data,
+          indexPatternService,
+          timefilter,
+          inspector,
+          expressionRenderer,
+          basePath: coreHttp.basePath,
+          getTrigger: uiActions?.getTrigger,
+          getTriggerCompatibleActions: uiActions?.getTriggerCompatibleActions,
+          documentToExpression,
+          injectFilterReferences,
+          visualizationMap,
+          datasourceMap,
+          capabilities: {
+            canSaveDashboards: Boolean(capabilities.dashboard?.showWriteControls),
+            canSaveVisualizations: Boolean(capabilities.visualize.save),
+            navLinks: capabilities.navLinks,
+            discover: capabilities.discover,
+          },
+          usageCollection,
+          theme,
+          spaces,
+          uiSettings,
+        },
+        input,
+        parent
+      );
+    } catch (e) {
+      return new ErrorEmbeddable(e, input, parent);
```

Comment:
This is definitely an improvement, but the downside is that you can't go to the Lens editor from the error embeddable - so a user wouldn't be able to fix a panel by themselves. Can we somehow make this possible? Can also be split out.

Reply:
https://github.com/elastic/kibana/issues/138608